azure/services/async/async.go

```diff
@@ -183,12 +194,27 @@ func (s *Service) DeleteResource(ctx context.Context, spec azure.ResourceSpecGet
 	return nil
 }
 
-// retryAfter returns the max between the `RETRY-AFTER` header and the default requeue time.
+// getRequeueAfterFromFuture returns the max between the `RETRY-AFTER` header and the default requeue time.
 // This ensures we respect the retry-after header if it is set and avoid retrying too often during an API throttling event.
-func retryAfter(sdkFuture azureautorest.FutureAPI) time.Duration {
+func getRequeueAfterFromFuture(sdkFuture azureautorest.FutureAPI) time.Duration {
 	retryAfter, _ := sdkFuture.GetPollingDelay()
 	if retryAfter < reconciler.DefaultReconcilerRequeue {
 		retryAfter = reconciler.DefaultReconcilerRequeue
 	}
 	return retryAfter
 }
+
+// getRetryAfterFromDetailedError returns the time.Duration from the http.Response in the autorest.DetailedError
+// If there is no Response object, or if there is no meaningful Retry-After header data, we return a zero value time.Duration
+func getRetryAfterFromDetailedError(detailedError autorest.DetailedError) (ret time.Duration) {
+	if detailedError.Response != nil {
+		if retryAfter := detailedError.Response.Header.Get("Retry-After"); retryAfter != "" {
+			if rai, _ := strconv.Atoi(retryAfter); rai > 0 {
+				ret = time.Duration(rai) * time.Second
+			}
+		} else if detailedError.Response.StatusCode == http.StatusTooManyRequests {
+			ret = 1 * time.Minute
+		}
+	}
+	return
```

Comment:
could we return the Default requeue time instead of 0 if no retry after is found? Like above https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2688/files#diff-0e657fbf13cf152e97cf8871a3baf550199d64f99f94316e7e1b9eeb5d6cc8e4R202

Reply:
So the idea here would be that if we receive a strongly typed `autorest.DetailedError` error, then we should always classify that as a transient error w/ default requeue?

Reply:
I've moved all of this into the `getRetryAfterFromError` func, and now the GET in the flow of `CreateResource` *always* returns a transient `azure.ReconcileError`.

Is that correct? That's a change from the present, where non-404 errors are always returned as vanilla errors (`err != nil && !azure.ResourceNotFound(err)`. Is there a chance that changing this to transient `azure.ReconcileError` will result in a mis-classification of non-transient errors?